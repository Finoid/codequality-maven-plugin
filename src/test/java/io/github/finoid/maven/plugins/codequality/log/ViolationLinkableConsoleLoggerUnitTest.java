package io.github.finoid.maven.plugins.codequality.log;

import io.github.finoid.maven.plugins.codequality.fixtures.UnitTest;
import io.github.finoid.maven.plugins.codequality.fixtures.ViolationFaker;
import org.junit.jupiter.api.Test;

class ViolationLinkableConsoleLoggerUnitTest extends UnitTest {
    private final ViolationLinkableConsoleLogger unit = new ViolationLinkableConsoleLogger();

    @Test
    void givenViolation_whenFormat_thenExpectedFormat() {
        var violation = ViolationFaker.violation()
            .create();

        var result = unit.format(violation);

        snapshot(result);
    }
}