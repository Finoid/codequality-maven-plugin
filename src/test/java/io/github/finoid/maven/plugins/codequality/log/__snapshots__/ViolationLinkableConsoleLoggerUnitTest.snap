io.github.finoid.maven.plugins.codequality.log.ViolationLinkableConsoleLoggerUnitTest.givenViolation_whenFormat_thenExpectedFormat=[
  "/path/to/file/bNnRLyKE.java:[72,17] Description Zv2g7O0N4P [ZsDtD78bWT]"
]